core/contractsdk/go/example/htlc/hashed_timelock.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"github.com/xuperchain/xuperchain/core/contractsdk/go/code"
+	"github.com/xuperchain/xuperchain/core/contractsdk/go/driver"
+	"math/big"
+	"strconv"
+)
+
+type HashedTimelock struct{}
+
+// riddleHash --> LockContract{...}
+type LockContract struct {
+	sender        string
+	receiver      string
+	amount        string
+	expiredHeight string
+	riddleHash    string
+}
+
+func (c *HashedTimelock) Initialize(ctx code.Context) code.Response {
+	return code.OK(nil)
+}
+
+//Open a riddle
+func (c *HashedTimelock) Open(ctx code.Context) code.Response {
+	receiver, ok := ctx.Args()["receiver"]
+	if !ok {
+		return code.Errors("missing receiver")
+	}
+	riddleHash, ok := ctx.Args()["riddle_hash"]
+	if !ok {
+		return code.Errors("missing riddle_hash")
+	}
+	oldValue, _ := ctx.GetObject(riddleHash)
+	if oldValue != nil {
+		return code.Errors("duplicated riddle hash:" + string(riddleHash))
+	}
+	expiredHeight, ok := ctx.Args()["expired_height"]
+	if !ok {
+		return code.Errors("missing expired_height")
+	}
+	sender := ctx.Initiator()
+	amount, _ := ctx.TransferAmount()
```

Comment:
Could amount be negative number or zero?

Reply:
It should be greater than zero. Actually, it is better to be higher than the tipBlock's height. 
And, I have added a check for args to prevent misstakes of users.

Reply:
yes, it would be better to check the range of amount here. thanks